### PR TITLE
[libc] Check if arm targets support FPSCR in FEnvImpl.h.

### PR DESCRIPTION
**Pull request by lntue**
https://github.com/llvm/llvm-project/issues/72157

### CODE REVIEW

File: libc/src/__support/FPUtil/FEnvImpl.h [hunk 1]
```diff
@@ -28,7 +28,7 @@
 // the apple condition here should be removed.
 #elif defined(LIBC_TARGET_ARCH_IS_X86) && !defined(__APPLE__)
 #include "x86_64/FEnvImpl.h"
-#elif defined(LIBC_TARGET_ARCH_IS_ARM)
+#elif defined(LIBC_TARGET_ARCH_IS_ARM) && __has_builtin(__builtin_arm_get_fpscr)
```
Comment (michaelrj-google) [inline libc/src/__support/FPUtil/FEnvImpl.h:31]: iirc `__has_builtin` is a clang specific extension, should we also check what compiler we're using?

Comment (lntue) [reply libc/src/__support/FPUtil/FEnvImpl.h]: gcc also has `__has_builtin` : https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fbuiltin.html
But I'm switching to use our wrapper macro `LIBC_HAS_BUILTIN` instead.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Comment by petrhosek** [comment]
We could use `defined(__ARM_FP)` which is set if hardware floating-point is available.

**Pull request merged**